pkg/networkservice/common/monitor/server.go

```diff
@@ -77,43 +85,80 @@ func (m *monitorServer) MonitorConnections(selector *networkservice.MonitorScope
 }
 
 func (m *monitorServer) Request(ctx context.Context, request *networkservice.NetworkServiceRequest) (*networkservice.Connection, error) {
+	// Pass metrics monitor, so it could be used later in chain.
+	ctx = WithServer(ctx, m)
+
+	connectionClone := proto.Clone(request.GetConnection())
 	conn, err := next.Server(ctx).Request(ctx, request)
 	if err == nil {
```

Comment:
Could you please rewrite code in the following way to avoid nesting?
```go
if err != nil {
    return nil, err
}

m.executor.AsyncExec(func() {
	m.connections[conn.GetId()] = conn

	if proto.Equal(connectionClone, conn) {
		return
	}

	if err := m.send(ctx, event); err != nil {
		logrus.Errorf("Error during sending event: %v", err)
	}
})

return conn, err

```